### PR TITLE
[staging-17.0] Merge master from 01d485569

### PR DESCRIPTION
**Pull request by darcywong00**
1 of x of merging master from 01d485569 into staging-17.0

When migrating info from .keyboard_info to .kps, I'm still a little unclear when a font becomes `<DisplayFont>` vs `<WebDisplayFont>`

### CODE REVIEW

File: release/n/newa_traditional_extended/source/newa_traditional_extended.kps [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  <System>
+    <KeymanDeveloperVersion>16.0.142.0</KeymanDeveloperVersion>
+    <FileVersion>7.0</FileVersion>
+  </System>
+  <Options>
+    <ExecuteProgram></ExecuteProgram>
+    <ReadMeFile>readme.htm</ReadMeFile>
+    <MSIFileName></MSIFileName>
+    <MSIOptions></MSIOptions>
+    <FollowKeyboardVersion/>
```
Comment (mcdurdin) [inline release/n/newa_traditional_extended/source/newa_traditional_extended.kps:12]:
```suggestion
    <FollowKeyboardVersion/>
    <LicenseFile>..\LICENSE.md</LicenseFile>
```

File: release/n/newa_traditional_extended/source/newa_traditional_extended.kps [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  <Info>
+    <Name URL="">Newa Traditional Extended</Name>
+    <Copyright URL="">© 2023 Swornim Nakarmi</Copyright>
+    <Author URL="">Swornim Nakarmi</Author>
+    <Description>Newa Traditional Extended: Extended Traditional Layout for Nepal Bhasa 
+in Prachalit Newa Script with the mostly used conjuncts and special characters.</Description>
+    <Version URL=""></Version>
+  </Info>
+  <Files>
+    <File>
```
Comment (mcdurdin) [inline release/n/newa_traditional_extended/source/newa_traditional_extended.kps:27]:
```suggestion
    <File>
      <Name>..\LICENSE.md</Name>
      <Description></Description>
      <CopyLocation>0</CopyLocation>
      <FileType>.md</FileType>
    </File>
    <File>
```

**Review by mcdurdin — CHANGES_REQUESTED** [review]
A few minor tweaks needed, cheers :grin:

File: release/b/burushaski_girminas_perso_arabic/source/burushaski_girminas_perso_arabic.kps [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    <KeymanDeveloperVersion>16.0.142.0</KeymanDeveloperVersion>
+    <FileVersion>7.0</FileVersion>
+  </System>
+  <Options>
+    <ExecuteProgram></ExecuteProgram>
+    <ReadMeFile>readme.htm</ReadMeFile>
+    <GraphicFile>icon_burushaski_keyboard_package.png</GraphicFile>
+    <MSIFileName></MSIFileName>
+    <MSIOptions></MSIOptions>
+    <FollowKeyboardVersion/>
```
Comment (mcdurdin) [inline release/b/burushaski_girminas_perso_arabic/source/burushaski_girminas_perso_arabic.kps:13]:
```suggestion
    <FollowKeyboardVersion/>
    <LicenseFile>..\LICENSE.md</LicenseFile>
```

File: release/n/newa_traditional_extended/source/newa_traditional_extended.kps [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      <CopyLocation>0</CopyLocation>
+      <FileType>.jpg</FileType>
+    </File>
+    <File>
+      <Name>..\..\..\shared\fonts\noto\Newa\NotoSansNewa-Regular.ttf</Name>
+      <Description>Font Noto Sans Newa Regular</Description>
+      <CopyLocation>0</CopyLocation>
+      <FileType>.ttf</FileType>
+    </File>
+  </Files>
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply release/n/newa_traditional_extended/source/newa_traditional_extended.kps:87]:
```suggestion
     <File>
      <Name>..\LICENSE.md</Name>
      <Description></Description>
      <CopyLocation>0</CopyLocation>
      <FileType>.md</FileType>
    </File>
  </Files>
```

File: release/b/burushaski_girminas/source/burushaski_girminas.kps [hunk 5]
```diff
@@ -69,21 +70,27 @@
...
-      <FileType>.md</FileType>
+      <FileType>.png</FileType>
+    </File>
+    <File>
+      <Name>welcome\image_keyboard_guide2.png</Name>
+      <Description>File image_keyboard_guide2.png</Description>
+      <CopyLocation>0</CopyLocation>
+      <FileType>.png</FileType>
     </File>
   </Files>
```
Comment (darcywong00) [inline release/b/burushaski_girminas/source/burushaski_girminas.kps:84]:
```suggestion
    <File>
      <Name>..\LICENSE.md</Name>
      <Description></Description>
      <CopyLocation>0</CopyLocation>
      <FileType>.md</FileType>
    </File>
  </Files>
```

File: release/b/burushaski_girminas_perso_arabic/source/burushaski_girminas_perso_arabic.kps [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      <CopyLocation>0</CopyLocation>
+      <FileType>.png</FileType>
+    </File>
+    <File>
+      <Name>welcome\fonts\Burusho_Quwatz_Nastaliq.ttf</Name>
+      <Description>Font Burusho Quwatz Nastaliq</Description>
+      <CopyLocation>0</CopyLocation>
+      <FileType>.ttf</FileType>
+    </File>
+  </Files>
```
Comment (darcywong00) [inline release/b/burushaski_girminas_perso_arabic/source/burushaski_girminas_perso_arabic.kps:89]:
```suggestion
    <File>
      <Name>..\LICENSE.md</Name>
      <Description></Description>
      <CopyLocation>0</CopyLocation>
      <FileType>.md</FileType>
    </File>
  </Files>
```

File: release/bj/bj_cree_east/source/bj_cree_east.kps [hunk 7]
```diff
@@ -91,9 +91,10 @@ Syllabics.</Description>
     <Keyboard>
       <Name>BJCreeUNI (east)</Name>
       <ID>bj_cree_east</ID>
-      <Version>9.1</Version>
-      <OSKFont>..\..\..\shared\fonts\cree\bjcrus.ttf</OSKFont>
+      <Version></Version>
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply release/bj/bj_cree_east/source/bj_cree_east.kps:94]:
```suggestion
      <Version>9.1</Version>
```

File: release/b/burushaski_girminas_perso_arabic/burushaski_girminas_perso_arabic.kpj [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+      <FileVersion></FileVersion>
+      <FileType>.md</FileType>
+    </File>
+    <File>
+      <ID>id_961ac9365d9878cbd825cc5f8fbeb68c</ID>
+      <Filename>burushaski_girminas_perso_arabic.keyboard_info</Filename>
+      <Filepath>burushaski_girminas_perso_arabic.keyboard_info</Filepath>
+      <FileVersion></FileVersion>
+      <FileType>.keyboard_info</FileType>
+    </File>
```
Comment (darcywong00) [inline release/b/burushaski_girminas_perso_arabic/burushaski_girminas_perso_arabic.kpj:61]: Removing keyboard_info from .kps

```suggestion
```

File: release/n/newa_traditional_extended/newa_traditional_extended.kpj [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+      <FileVersion></FileVersion>
+      <FileType>.md</FileType>
+    </File>
+    <File>
+      <ID>id_10c53d9c855ecf8db7023986675888c4</ID>
+      <Filename>newa_traditional_extended.keyboard_info</Filename>
+      <Filepath>newa_traditional_extended.keyboard_info</Filepath>
+      <FileVersion></FileVersion>
+      <FileType>.keyboard_info</FileType>
+    </File>
```
Comment (darcywong00) [inline release/n/newa_traditional_extended/newa_traditional_extended.kpj:61]:
```suggestion
```
Removing keyboard_info from kps

**Review by mcdurdin — APPROVED** [review]

**Comment by mcdurdin** [comment]
> When migrating info from .keyboard_info to .kps, I'm still a little unclear when a font becomes `<DisplayFont>` vs `<WebDisplayFont>`

If .keyboard_info has a `"font"` section in a `"languages"` entry, then that turns into `<WebDisplayFonts>` in the .keyboard_info, and `"oskFont"` becomse `<WebOSKFonts>`. e.g.:

```json
  "languages": {
    "km": {
      "oskFont": {
        "family": "Khmer Busra Kbd",
        "source": [
          "khmer_busra_osk.ttf"
        ]
      },
```

becomes

```xml
<WebOSKFonts><Font Filename="khmer_busra_osk.ttf" /></WebOSKFonts>
```

`<DisplayFont>` and `<OSKFont>` data in a .kps file should not be impacted by the .keyboard_info import; these have already been defined for a few versions.

The reason the data is split out for web fonts is so that we can list multiple alternative fonts, e.g. .woff, .woff2, which the existing DisplayFont and OSKFont elements did not support.

**Comment by darcywong00** [comment]
> > When migrating info from .keyboard_info to .kps, I'm still a little unclear when a font becomes `<DisplayFont>` vs `<WebDisplayFont>`
> 
> If .keyboard_info has a `"font"` section in a `"languages"` entry, then that turns into `<WebDisplayFonts>` in the .keyboard_info, and `"oskFont"` becomse `<WebOSKFonts>`. e.g.:
> 
> ```json
>     "km": {
>       "oskFont": {
>         "family": "Khmer Busra Kbd",
>         "source": [
>           "khmer_busra_kbd.ttf"
>         ]
>       },
>       "font": {
>         "family": "Khmer Mondulkiri",
>         "source": [
>           "Mondulkiri-R.ttf"
>         ]
>       },
> ```
> 
> becomes
> 
> ```
> <WebDisplayFonts><Font Filename="Mondulkiri-R.ttf" /></WebDisplayFonts>
> <WebOSKFonts><Font Filename="khmer_busra_osk.ttf" /></WebOSKFonts>
> ```
> 
> `<DisplayFont>` and `<OSKFont>` data in a .kps file should not be impacted by the .keyboard_info import; these have already been defined for a few versions.
> 
> The reason the data is split out for web fonts is so that we can list multiple alternative fonts, e.g. .woff, .woff2, which the existing DisplayFont and OSKFont elements did not support.

gotcha
Last commit https://github.com/keymanapp/keyboards/pull/2734/commits/20d3f4289ae9956892f2192d59efac1c064304e8 was to add newa_traditional_extended `<WebDisplayFonts>` from the (removed) keyboard_info font.

**Pull request merged**